### PR TITLE
Update Model3DTileContent specs to have parity with 3D Tiles 1.0 formats

### PR DESCRIPTION
**Pull request by ptrgags**
This PR adds specs from `Batched3DModel3DTileContent`, `Instanced3DModel3DTileContent` and `PointCloud3DTileContent` to `Model3DTileContent` (and sometimes other places like the loaders)

Some notes:

* I discovered that `Model3DTileContent.getFeature()` wasn't throwing exceptions like the interface expects, so I updated the code there
* The point cloud back face normals is a little more involved than I first thought so I `xit()`-ed the test and opened https://github.com/CesiumGS/cesium/issues/10632 so we don't have to hold up the staging branch.

@j9liu could you review?

### CODE REVIEW

**Comment by cesium-concierge** [comment]
Thanks for the pull request @ptrgags!
* :heavy_check_mark: Signed CLA found.

Reviewers, don't forget to make sure that:

- [ ] Cesium Viewer works.
- [ ] Works in 2D/CV.

File: Specs/Scene/ModelExperimental/Model3DTileContentSpec.js [hunk 1]
```diff
@@ -464,73 +924,431 @@ describe(
...
-    it("picks from b3dm batch table", function () {
-      setCamera(centerLongitude, centerLatitude, 15.0);
-      return Cesium3DTilesTester.loadTileset(scene, withBatchTableUrl).then(
-        function (tileset) {
+    describe("glTF", function () {
+      beforeEach(function () {
+        setCamera(centerLongitude, centerLatitude, 100.0);
+      });
+
+      it("resolves readyPromise with glb", function () {
```
Comment (j9liu) [inline Specs/Scene/ModelExperimental/Model3DTileContentSpec.js:1271]: is there any reason these stayed in but the `"resolves readyPromise with b3dm"` and `"resolves readyPromise with i3dm"` tests got removed?

File: Specs/Scene/ModelExperimental/Model3DTileContentSpec.js [hunk 2]
```diff
@@ -545,96 +1363,245 @@ describe(
             expect(content.hasProperty(featureId, "id")).toBe(true);
             expect(content.getFeature(featureId)).toBeDefined();
           });
-        }
-      );
+        });
+      });
+
+      // only have one destroy test (this one)
+      it("destroys glTF content", function () {
```
Comment (j9liu) [inline Specs/Scene/ModelExperimental/Model3DTileContentSpec.js:1370]: FYI there's a "destroy pnts content" test, so either this comment or that test should be removed to be consistent

Comment (j9liu) [reply Specs/Scene/ModelExperimental/Model3DTileContentSpec.js:1370]: And if we're keeping this comment, maybe reword this more formally...
```suggestion
      // Since all content types are loaded as a Model3DTileContent, only one destroy test is necessary
      it("destroys", function () {
```
... and move the test outside the `glTF` suite

File: Source/Scene/ModelExperimental/Model3DTileContent.js [hunk 3]
```diff
@@ -142,10 +143,27 @@ Model3DTileContent.prototype.getFeature = function (featureId) {
...
+  if (!defined(featureTable)) {
+    throw new DeveloperError(
+      "No feature table found for the selected feature ID set"
+    );
+  }
+
+  const featuresLength = featureTable.featuresLength;
+  if (!defined(featureId) || featureId < 0 || featureId >= featuresLength) {
+    throw new DeveloperError(
+      `featureId is required and between 0 and featuresLength - 1 (${
```
Comment (j9liu) [inline Source/Scene/ModelExperimental/Model3DTileContent.js:162]:
```suggestion
      `featureId is required and must be between 0 and featuresLength - 1 (${
```

**Review by j9liu — CHANGES_REQUESTED** [review]
@ptrgags left some comments, let me know if you have any questions

File: Specs/Scene/ModelExperimental/Model3DTileContentSpec.js (hunk 2 above)
Comment (ptrgags) [reply Specs/Scene/ModelExperimental/Model3DTileContentSpec.js]: ah I forgot to put `TODO` in the comment so I missed a spot. 

This comment was saying keep the glTF destroy test, but remove the pnts/b3dm/i3dm equivalents. I updated these.

File: Specs/Scene/ModelExperimental/PntsLoaderSpec.js [hunk 4]
```diff
@@ -13,6 +14,7 @@ import {
   VertexAttributeSemantic,
 } from "../../../Source/Cesium.js";
 import createScene from "../../createScene.js";
+import pollToPromise from "../../pollToPromise.js";
```
[the remark this answers is not in the document]
Comment (ptrgags) [reply Specs/Scene/ModelExperimental/PntsLoaderSpec.js:17]: Is there a reason why this comment is in `PntsLoaderSpec`?

for the tile transform tests, we only wanted to have one set of them since it works the same for each content type. For this one I kept the `b3dm` ones since I was having difficulty getting the camera setup right for the glTF ones.

File: Specs/Scene/ModelExperimental/Model3DTileContentSpec.js (hunk 1 above)
Comment (ptrgags) [reply Specs/Scene/ModelExperimental/Model3DTileContentSpec.js:1254]: this was one of the cases where we wanted to consolidate the test to only use one instead of checking every format. So I just kept the glTF ones.

**Comment by ptrgags** [comment]
@j9liu updated everything except for a couple questions I left in the comment threads

File: Specs/Scene/ModelExperimental/PntsLoaderSpec.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (j9liu) [reply Specs/Scene/ModelExperimental/PntsLoaderSpec.js:17]: > Is there a reason why this comment is in `PntsLoaderSpec`?

Because most of the tilesets tested in `Point3DTileContentSpec` were copied over to `PntsLoaderSpec`, except for the one with the tile transform. So if we don't duplicate it, the tileset associated with it will never get tested

File: Specs/Scene/ModelExperimental/Model3DTileContentSpec.js (hunk 1 above)
Comment (j9liu) [reply Specs/Scene/ModelExperimental/Model3DTileContentSpec.js:1255]: okay thats fine. I was just confused by the specificity between glTF and glb

File: Specs/Scene/ModelExperimental/Model3DTileContentSpec.js [hunk 5]
```diff
@@ -545,96 +1363,245 @@ describe(
...
-        expect(scene).toPickAndCall(function (result) {
-          expect(result).toBeDefined();
-          expect(result.primitive).toBe(tileset);
-          expect(result.content).toBe(content);
-          const featureId = result.featureId;
-          expect(featureId).toBe(expectedFeatureId);
-          const feature = content.getFeature(featureId);
-          expect(feature).toBeDefined();
+    describe("clipping planes", function () {
+      const withBatchTableUrl =
```
[the remark this answers is not in the document]
Comment (j9liu) [reply Specs/Scene/ModelExperimental/Model3DTileContentSpec.js]: @ptrgags did you decide not to update this?

[the remark this answers is not in the document]
Comment (ptrgags) [reply Specs/Scene/ModelExperimental/Model3DTileContentSpec.js]: OH in my end-of-day haste I misread this, I thought you were talking about how the URLs were getting copied/pasted. 

I'll update this today to use just the b3dm version if possible

File: Specs/Scene/ModelExperimental/PntsLoaderSpec.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (ptrgags) [reply Specs/Scene/ModelExperimental/PntsLoaderSpec.js:18]: Ah I understand now. I added a test in `PntsLoaderSpec`, though it seems of limited use, since the transform is applied in the `tileset.json`, not the `.pnts` file itself.

**Comment by ptrgags** [comment]
@j9liu thanks for answering my questions, I updated the PR!

File: Specs/Scene/ModelExperimental/PntsLoaderSpec.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (j9liu) [reply Specs/Scene/ModelExperimental/PntsLoaderSpec.js:18]: @ptrgags ah I see... let's just leave it there for now so we don't hold up the PR, we can move things around later.

**Comment by j9liu** [comment]
@ptrgags looks good, merging now!
